Adding VSCode tasks and updating appveyor

swernli opened the pull request:
@jterry75 @darrenstahlmsft 
Added VSCode tasks for build, test, and restore, as well as an empty launch file to keep VSCode happy.  Also added appveyor build of all projects to include samples and tests in CI.

Updated tests to use nanoserver by default.

swernli committed:
Adding VSCode tasks and updating appveyor

Added VSCode tasks for build, test, and restore, as well as an empty launch file to keep VSCode happy.  Also added appveyor build of all projects to include samples and tests in CI.

Updated tests to use nanoserver by default.

swernli commented:
Ok, this finally worked!  @darrenstahlmsft ready for review.

jterry75 commented on test/Microsoft.Windows.ComputeVirtualization.Tests/project.json:
```diff
@@ -10,11 +10,11 @@
   },
   "runtimes": {
-    "win10-x64": {}
+    "win81-x64": {}
```
How are we going to remember to revert this when AppVeyor is updated?

swernli replied:
I don't think we have a good way.  JSON does not support comments, so we can't add a reminder.

jterry75 replied:
Lets just file a task then. We can leave it open with a reminder to eventually look at it again.

jterry75 commented on test/Microsoft.Windows.ComputeVirtualization.Tests/ComputeTests.cs:
```diff
@@ -13,7 +13,7 @@ public class ServercoreContainerTests : IDisposable
         public ServercoreContainerTests()
         {
-            sandbox = new Sandbox(Sandbox.ContainerType.ServerCore, id);
+            sandbox = new Sandbox(Sandbox.ContainerType.NanoServer, id);
```
Odd that the class is "ServerCoreContainerTests" but you changed this to "NanoServer"

darrenstahlmsft commented:
I think global.json should include the test directory in order for VsCode to find it.

swernli commented:
Updated the global.json, and added a build task for the tests.
Also, made a change to the test class name to make it generic and changed the two environment variables to be one.  Test cases can distinguish between the two types of containers, but we really shouldn't need to differentiate in the test code between OSes... that's the responsibility of the entity launching the tests.

darrenstahlmsft commented:
LGTM
